### PR TITLE
Document ./x.py fmt

### LINKED ISSUE #577: Document fmt in Other Tests
It would be nice to document that we are testing the code is fmted here https://rust-lang.github.io/rustc-guide/tests/intro.html#other-tests

### PR DESCRIPTION
**Pull request by mark-i-m**
fixes #577

### CODE REVIEW

File: src/tests/intro.md [hunk 1]
```diff
@@ -63,6 +63,15 @@ including:
 
   Example: `./x.py test src/tools/tidy`
 
+- **Formatting** – Rustfmt is integrated with the build system to enforce
+  uniform style across the compiler. In the CI, we check that the formatting
+  is correct.
```
Comment (LeSeulArtichaut) [inline src/tests/intro.md]: Should it be included that formatting is part of the tidy checks?

**Comment by spastorino** [comment]
I think after addressing what @LeSeulArtichaut have said this is good to be merged.

**Comment by mark-i-m** [comment]
@LeSeulArtichaut @spastorino Thanks :) I added a sentence mentioning that tidy runs a format check automatically.

**Comment by JohnTitor** [comment]
One note: `./x.py test src/tools/tidy --bless` does formatting before checking style, I think it's worth to mention it.

**Comment by mark-i-m** [comment]
@JohnTitor I added an example about that.

**Pull request merged**